### PR TITLE
Build warnings fix

### PR DESCRIPTION
**Pull request by baulus**

### CODE REVIEW

**Review by delliott90 — APPROVED** [review]

**Comment by baulus** [comment]
All warnings are fixed except the following:
```yaml
virtualenv/lib/python3.9/site-packages/botocore/httpsession.py:41
  /Users/amuradyan/Projects/stix-shifter/opencybersecurityalliance_async/stix-shifter/virtualenv/lib/python3.9/site-packages/botocore/httpsession.py:41: DeprecationWarning: 'urllib3.contrib.pyopenssl' module is deprecated and will be removed in a future release of urllib3 2.x. Read more in this issue: https://github.com/urllib3/urllib3/issues/2680
    from urllib3.contrib.pyopenssl import orig_util_SSLContext as SSLContext
```

```yaml
../../../../../opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/abc.py:106
  /opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/abc.py:106: DeprecationWarning: Inheritance class AiohttpSession from ClientSession is discouraged
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
```

We could have try solving this error by upgraded [botocore lib](https://pypi.org/project/botocore/), but the current [aiobotocore lib requires](https://github.com/aio-libs/aiobotocore/blob/master/setup.py) rigid version of botocore, which is >=1.27.59,<1.27.60

File: stix_shifter/requirements.txt [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 flatten_json==0.1.13
+json-fix==0.5.1
+jsonmerge==1.9.0
+pyOpenSSL==23.0.0
 python-dateutil==2.8.2
 requests_toolbelt==0.10.1
+stix2-matcher==3.0.0
 stix2-patterns==1.3.2
-stix2-validator==3.0.2
+stix2-validator==3.1.3
```
Comment (baulus) [inline stix_shifter/requirements.txt:18]: @delliott90 @mdazam1942 do we have a concern using this latest version of stix2-validator?

Comment (mdazam1942) [reply stix_shifter/requirements.txt:18]: we never tested actually if existing mappings passes the validation with the new validator. If you can test then i guess we can update this version @baulus

Comment (baulus) [reply stix_shifter/requirements.txt:18]: test with
```bash
% python main.py execute "stix_bundle" "stix_bundle" '{"type": "identity", "id": "identity--f431f809-377b-45e0-aa1c-6a4751cae5ff", "name": "stix_bundle", "identity_class": "events"}' '{"url": "https://raw.githubusercontent.com/opencybersecurityalliance/stix-shifter/develop/data/cybox/1.json", "options": {"stix_validator": true, "stix_2.1": false}}' '{"auth": {"username": null, "password": null}}' "[ipv4-addr:value = '127.0.0.1']  START t'2019-03-01T10:57:51.826Z' STOP t'2023-03-01T11:02:51.826Z'"
```
with "stix_validator": true/false and "stix_2.1": false/true for 
https://raw.githubusercontent.com/opencybersecurityalliance/stix-shifter/develop/data/cybox/stix-2.1.json
and 
https://raw.githubusercontent.com/opencybersecurityalliance/stix-shifter/develop/data/cybox/1.json

**Review by mdazam1942 — APPROVED** [review]

**Pull request merged**